### ISSUE TITLE
Lookup.searchitem unable to search item as select lookup opens the flyout

### ISSUE THREAD

**Issue by sudiptosahoo** [Microsoft/EasyRepro#111]
Lookup.searchitem("item to search") failed to search the item, as select lookup opens the lookup fly out. 
xrmBrowser.Entity.SelectLookup("lookupID");
xrmBrowser.Loookup.SearchItem("search text");

failed as select lookup opens the lookup flyout, failing all testcases containing lookup.

Impacted areas/test cases: webclient test cases (8.2 branch) having lookup.

**Comment by TYLEROL** [comment]
Thanks for raising this issue @sudiptosahoo. I am reviewing this reported issue and will report back as soon as I have more information to share.

Regards,
Tyler

**Comment by TYLEROL** [comment]
@sudiptosahoo - I have attempted test iterations of this on both the 8.2 and 9.0 branches using Dynamics 365 organizations and have not been able to reproduce a failure selecting the lookup flyout.

I tested both Lookup.Search methods with and without wait times (example below has a wait time). Both worked successfully.

If you're still having issues, can you please provided the following:

- Browser & Browser Version
- Browser Driver version (e.g. ChromeDriver v2.37)
- Exception detail during the failed test
- Code from your failed test (can test on our side)
- Does this fail whether the targeted lookup contains data or not?

Thanks,
Tyler

```
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Microsoft.Dynamics365.UIAutomation.Api;
using Microsoft.Dynamics365.UIAutomation.Browser;
using System;
using System.Security;
using System.Collections.Generic;
using OpenQA.Selenium;

namespace Microsoft.Dynamics365.UIAutomation.Sample
{
    [TestClass]
    public class OpenAccount
    {

        private readonly SecureString _username = System.Configuration.ConfigurationManager.AppSettings["OnlineUsername"].ToSecureString();
        private readonly SecureString _password = System.Configuration.ConfigurationManager.AppSettings["OnlinePassword"].ToSecureString();
        private readonly Uri _xrmUri = new Uri(System.Configuration.ConfigurationManager.AppSettings["OnlineCrmUrl"].ToString());

        private readonly SecureString _tieusername = System.Configuration.ConfigurationManager.AppSettings["TieOnlineUsername"].ToSecureString();
        private readonly SecureString _tiepassword = System.Configuration.ConfigurationManager.AppSettings["TieOnlinePassword"].ToSecureString();
        private readonly Uri _tiexrmUri = new Uri(System.Configuration.ConfigurationManager.AppSettings["TieOnlineCrmUrl"].ToString());

        [TestMethod]
        public void TestOpenActiveAccount()
        {
            using (var xrmBrowser = new XrmBrowser(TestSettings.Options))
            {
                xrmBrowser.LoginPage.Login(_xrmUri, _username, _password);
                xrmBrowser.ThinkTime(10000);
                xrmBrowser.Navigation.OpenSubArea("Sales", "Accounts");
                xrmBrowser.ThinkTime(2000);
                xrmBrowser.Grid.SwitchView("Active Accounts");
                xrmBrowser.ThinkTime(1000);
                //Open Primary Entity record (Account)
                xrmBrowser.Grid.OpenRecord(0);

                xrmBrowser.Entity.SelectLookup("parentaccountid");

                xrmBrowser.Lookup.Search("Test", 5000); //Included a 5 second wait on the Lookup Search

            }
        }
    }
}
```

**Comment by TYLEROL** [comment]
@sudiptosahoo - Please let us know if you're still encountering issues with this test and/or the sample I provided above.

We also released an updated version of the APIs last night, so I would encourage you to test with those if your issues persist in the prior version. If the latest version has not resolve your problems, please re-open this issue for further review.

Regards,
Tyler

**Comment by sudiptosahoo** [comment]
Yes I am still unable to use lookup.search directly

                xrmBrowser.Entity.SelectLookup("ttc_vehicalroute");
                xrmBrowser.Lookup.Search("35 JANE");
this code snippet not working.

Thanks,
Sudipto

**Comment by mkmk89** [comment]
Hi, This works for me when Lookup field is empty. When it has value, the SelectLookup clicks field first, which opens related records, which fails the process.

**Comment by TYLEROL** [comment]
@mkmk89 - Thank you for also validating that you're seeing issues with the SelectLookup method when the field contains data.

I tested this scenario again today and I am now able to reproduce this behavior. We're working on a fix and will update as soon as one is made available.

Regards,
Tyler

**Comment by aarti-ubriani** [comment]
I am not able to select a value with the selectLookup method as well. Any idea when this would be fixed ?

**Comment by tipsey** [comment]
Using v9.0 branch on On-premise solution and also having problem selecting lookup value. It always selects the first item in the lookup list:  ie.  Lookup list is in this order:  Phone Call, Voicemail, Email, Other

                xrmBrowser.QuickCreate.SelectLookup("caseorigincode", "Email");

                xrmBrowser.QuickCreate.SelectLookup("caseorigincode", 2);

Result:  It always populates the field with "Phone Call" instead of "Email".  Any workarounds to this?  Thank you.

**Comment by TYLEROL** [comment]
@sudiptosahoo, @mkmk89  and @aarti-ubriani  - Could you re-test this scenario with the latest code from the beta/9.0.2 branch on GitHub?

I've tested this scenario on v8.2.2 and v9.0.2 when the Lookup both contains and does not contain data, and I no longer receive any exceptions. In both of these scenarios, the Lookup flyout opens as expected. I am then able to call the Lookup.Search() method and it opens the Lookup dialog and enters the search text. In the scenario where data was present, we clear the value and enter the desired text.

I am closing the issue at this time, however please comment and/or re-open the issue if you're still observing problems. **If you do encounter issues, please post your code, the exception received, and the Browser and related Browser Driver being used.**


Best Regards,
Tyler

**Comment by TYLEROL** [comment]
@sudiptosahoo , @mkmk89 , and @aarti-ubriani - As recently mentioned in Issue #191, I described the usage of the Lookup.Search() method incorrectly.

After selecting the lookup, and performing the search, the Lookup.Add() method must be called to add the item via the dialog window. If you have multiple results in the search dialog and want a result _other than the first result,_ you can choose the result using Lookup.SelectItem(_indexvalue_) method. 

I've included a sample below:

```
        public void WEBTestCreateNewContact()
        {
            using (var xrmBrowser = new Api.Browser(TestSettings.Options))
            {
                xrmBrowser.LoginPage.Login(_xrmUri, _username, _password);
                xrmBrowser.GuidedHelp.CloseGuidedHelp();

                xrmBrowser.ThinkTime(500);
                xrmBrowser.Navigation.OpenSubArea("Sales", "Contacts");

                xrmBrowser.ThinkTime(1000);
                xrmBrowser.Grid.SwitchView("Active Contacts");

                xrmBrowser.ThinkTime(1000);
                xrmBrowser.CommandBar.ClickCommand("New");

                xrmBrowser.ThinkTime(1000);

                xrmBrowser.Entity.SelectLookup("parentcustomerid");

                xrmBrowser.Lookup.Search("C", 2000); //Included a 2 second wait on the Lookup Search

                //By default, the first record should always be selected. SelectItem() method should only be required if there is more than a single result.
                //xrmBrowser.Lookup.SelectItem(1); 

                xrmBrowser.Lookup.Add();

                xrmBrowser.CommandBar.ClickCommand("Save");
                
            }
        }
```

I'll review the SelectLookup() method again when the lookup field contains data to verify that the field is cleared first.

If you have any issues with the approach described above, please let me know.

Best Regards,
Tyler

**Comment by TYLEROL** [comment]
@mkmk89 - I've pushed an additional PR this morning #194 which should resolve the issue you faced when the lookup field already contained a value.

This PR includes logic so that in the event you WANT to select the field value and open the record from the lookup, you can pass 'false' for the _clearFieldValue_ parameter. By default this is set to True.

`                xrmBrowser.Entity.SelectLookup("parentcustomerid", false);`

If you face any additional issues, please open a new issue ticket and we'll review accordingly.

Best Regards,
Tyler

**Comment by SachchinAnnam** [comment]
Thanks Tyler, it worked :)
